### PR TITLE
Introduce ability to generate RC RPMs with a minimal pipeline

### PR DESCRIPTION
**Pull request by soumyadeep2007**
This PR makes it possible to generate RC RPMs with a minimal pipeline
(without test jobs) in order to test an external product which tests against RC
RPMs, such as gpupgrade. This would ensure that we can generate a RPM against a
dev branch of GPDB to test against gpupgrade.

Please refer to the individual commits for more details.

TODO: Set prod pipeline after merging this (even though there are no semantic
changes, to make state-checker happy)

Sample pipeline generated with these changes:
https://cm.ci.gpdb.pivotal.io/teams/main/pipelines/master_rc_test

This needs to be backported to 5X and 6X

Co-authored-by: Kalen Krempely <kkrempely@vmware.com>

### CODE REVIEW

File: concourse/pipelines/gen_pipeline.py [hunk 1]
```diff
@@ -172,7 +172,9 @@ def create_pipeline(args):
         'pipeline_configuration': args.pipeline_configuration,
         'test_trigger': test_trigger,
         'use_ICW_workers': args.use_ICW_workers,
-        'build_test_rc_rpm': args.build_test_rc_rpm
+        'build_test_rc_rpm': args.build_test_rc_rpm,
+        'git_user': git_remote.split('/')[-2],
```
Comment (kalensk) [inline concourse/pipelines/gen_pipeline.py:176]: nit: Consider renaming to git_username instead of just user?

Comment (soumyadeep2007) [reply concourse/pipelines/gen_pipeline.py]: Done.

File: concourse/pipelines/gen_pipeline.py [hunk 2]
```diff
@@ -338,7 +335,7 @@ def main():
             'AA',
             'Extensions'
         ],
-        default=['ICW'],
+        default=[],
```
Comment (kalensk) [inline concourse/pipelines/gen_pipeline.py:338]: I'll let others chime in, but hopefully changing the default from ICW to just "compile" is "okay".

**Review by bmdoil — APPROVED** [review]
With @bhanu-atturu +1

**Comment by kalensk** [comment]
Thanks for the review!  : )

File: concourse/pipelines/templates/gpdb-tpl.yml [hunk 3]
```diff
@@ -1980,55 +1985,65 @@ jobs:
...
+          path: sh
+          args:
+            - -exc
+            - |
+              gpdb_semver=$(gpdb_src/getversion | cut -d' ' -f1)
+{% if build_test_rc_rpm %}
+              cp -v gpdb_rpm_centos7/*.rpm renamed_gpdb_rpm_centos7/greenplum-db-[[ git_username ]]-[[ git_branch ]]-${gpdb_semver}-rhel7-x86_64${RC_BUILD_TYPE_GCS}.rpm
+{% else %}
+              cp -v gpdb_rpm_centos7/*.rpm renamed_gpdb_rpm_centos7/greenplum-db-${gpdb_semver}-rhel7-x86_64${RC_BUILD_TYPE_GCS}.rpm
+{% endif %}
```
Comment (kalensk) [inline concourse/pipelines/templates/gpdb-tpl.yml:2041]: nit: consider a comment explaining why we rename the rpms using the git username and branch.

**Pull request merged**